OpenEdXMobile/src/main/java/org/edx/mobile/module/prefs/UserPrefs.java

```diff
@@ -46,18 +50,35 @@ public boolean isDownloadOverWifiOnly() {
         return onlyWifi;
     }
 
+    public boolean isSDCardDownloadEnabled(){
+        if (!environment.getConfig().isSDCardDownloadEnabled()){
+            return false;
+        }
+        final PrefManager prefManger = new PrefManager(context, PrefManager.Pref.SD_CARD);
+        return prefManger.getBoolean(PrefManager.Key.DOWNLOAD_TO_SDCARD, false);
+    }
+
     /**
-     * Returns user storage directory under /Android/data/ folder for the currently logged in user.
+     * Returns user storage directory under /Android/data/ folder for the currently logged in user
+     * or if the sd-card download is enabled the sd-card data location will be used.
      * This is the folder where all video downloads should be kept.
      *
      * @return
      */
     @Nullable
     public File getDownloadDirectory() {
-        final File externalAppDir = FileUtil.getExternalAppDir(context);
+        File downloadDir = null;
```

Comment:
It make more sense to move this function to FileUtil class.
Lets do it in this PR.